onnxruntime/contrib_ops/cpu/transformers/generation_shared.h

```diff
@@ -118,7 +119,13 @@ struct IBeamScorer {
                         Tensor* output_sequences,
                         Tensor* output_sequence_scores) = 0;
 
-  virtual gsl::span<int32_t>& GetNextIndices() = 0;
+  virtual bool IsDone() const = 0;                    // GPU version will return false here, as it asynchronously queues up the event
+  virtual bool IsDoneLater() const { return false; }  // GPU version waits for the asynchous result to complete here
```

Comment:
How about merge() IsDone and IsDoneLater() into one function IsDone(bool sync)?

For CUDA, sync=True will trigger synchronize. IsDone shall return true as long as the not-done count is zero no matter sync or not.

Reply:
I might not understand your suggestion fully, but I agree the names are not ideal. Hopefully the comments are okay.